experimenter/experimenter/nimbus_ui_new/views.py

```diff
@@ -252,11 +270,71 @@ def post(self, *args, **kwargs):
         return response
 
 
+class NimbusExperimentsCloneView(NimbusExperimentViewMixin, RequestFormMixin, UpdateView):
+    def get_form_kwargs(self):
+        kwargs = super().get_form_kwargs()
+        kwargs["data"] = kwargs["data"].copy()
+        kwargs["data"]["owner"] = self.request.user
+        return kwargs
+
+    def get_context_data(self, **kwargs):
+        context = super().get_context_data(**kwargs)
+        context["experiment"] = self.get_object()
+        return context
+
+    def post(self, *args, **kwargs):
+        response = super().post(*args, **kwargs)
+        if response.status_code == 302:
+            response = HttpResponse()
+            response.headers["HX-Redirect"] = reverse(
+                "nimbus-new-detail", kwargs={"slug": self.object.slug}
+            )
+        return response
+
+
+class NimbusExperimentsSidebarCloneView(NimbusExperimentsCloneView):
+    form_class = NimbusExperimentSidebarCloneForm
+    template_name = "nimbus_experiments/clone.html"
+
+
+class NimbusExperimentsPromoteToRolloutView(NimbusExperimentsCloneView):
+    form_class = NimbusExperimentPromoteToRolloutForm
+    template_name = "nimbus_experiments/clone.html"
+
+
+class UpdateCloneSlugView(NimbusExperimentViewMixin, RenderResponseMixin, UpdateView):
+    template_name = "nimbus_experiments/clone_slug_field.html"
+
+    def post(self, request, *args, **kwargs):
+        name = self.request.POST.get("name", "")
+        slug = slugify(name)
+        return self.render_to_response({"slug": slug})
+
+
+class ToggleArchiveView(
+    NimbusExperimentViewMixin,
+    RequestFormMixin,
+    RenderResponseMixin,
+    UpdateView,
+):
+    form_class = ToggleArchiveForm
+    template_name = "nimbus_experiments/archive_button.html"
+
+    def form_valid(self, form):
+        form.save()
+
+        if self.request.headers.get("HX-Request"):
+            response = HttpResponse()
+            response.headers["HX-Refresh"] = "true"
+            return response
```

Comment:
This should return an HttpResponse regardless.
```suggestion
    def form_valid(self, form):
        form.save()
        
        response = HttpResponse()
        if self.request.headers.get("HX-Request"):
            response.headers["HX-Refresh"] = "true"

        return response
```